debian/Cura.1
.\" Text automatically generated by txt2man
.TH Cura 2.3.1 "March 2017" "Cura 2.3.1" "User Commands"
.SH NAME
\fBCura \fP- Graphical user interface for the Cura slicing engine
.SH USAGE
Cura
.SH DESCRIPTION
Cura prepares 3D models for 3D printing.
.PP
It is the main GUI client for the Cura slicing engine.
.PP
For novices, it makes it easy to get great results.
.PP
For experts, there are over 200 settings to adjust to your needs.
.PP
It doesn't take any command line arguments.
.SH AUTHOR
Cura is Copyright (c) 2015-2017 \fBUltimaker B.V.\fR
.PP
This man page was written by \fBGregor Riepl\fR <\&onitake@gmail\&.com\&>.

CHANGES
Cura 15.06 Beta
===============

This is the *Beta* version of Cura 15.06.

Cura 15.06 is a new release built from the ground up on a completely new
framework called Uranium. This framework has been designed to make it easier to
extend Cura with additional functionality as well as provide a cleaner UI.

Changes since 15.05.95
----------------------

* Fixed: Selection ghost remains visible after deleting an object
* Fixed: Window does not show up immediately after starting application on OSX
* Fixed: Added display of rotation angle during rotation
* Fixed: Object changes position while rotating/scaling
* Fixed: Loading improvements in the layer view
* Fixed: Added application icons
* Fixed: Improved feedback when loading models
* Fixed: Eject device on MacOSX now provides proper feedback
* Fixed: Make it possible to show retraction settings for UM2
* Fixed: Opening the machine preferences page will switch to the first available machine
* Fixed: Improved tool handle hit area size
* Fixed: Render lines with a thickness based on screen DPI

Changes since 15.05.94
----------------------

* Added Russian translations
* Fixed: Infill not displayed in layer view
* Fixed: Cannot select/scale/rotate when first activating the tool and then trying to select a model.
* Fixed: Improved font rendering on Windows
* Fixed: Help > Show Documentation crashes Cura on Windows
* Fixed: "There is no disk in the drive" repeating messages on Windows
* Fixed: Retraction settings not visible for Ultimaker2
* Fixed: Display rotation angle when rotating an object
* Fixed: Time/Quality slider values are properly rounded
* Fixed: Improved clarity of buttons and text
* Fixed: No indication that anything is happening when loading a model
* Fixed: Eject device now works on Windows

Changes since 15.05.93
----------------------

* Fixed: No shortcuts for moving up/down layers in layer view.
* Fixed: Last view layers could not be scrolled through in layer view.
* Fixed: Files provided on command line would not actually show up on the build
  platform.
* Fixed: Render a ghost of the selection in Layer view to make the actual object
  position clear.
* Fixed: Showing a menu would clear the selection.
* Fixed: Size and scaling factor display for scale tool.
* Fixed: Missing background for additional tool controls.
* Fixed: Loading message times out when loading large files.
* Fixed: Show recent files in the file menu.
* Fixed: Windows installer will now install MSVC 2010 redistributable, to
  prevent issues with missing DLL's.
* Fixed: Collapsed/expanded state of setting categories not stored.

Changes since 15.05.91
----------------------

* There is now a working MacOSX version. Currently it supports OSX 10.7 and
  higher.
* Fixed: Need to deselect before selecting a different object.
* Fixed: Object can be moved on Z axis.
* Fixed: Error values should be considered invalid values and will not trigger a
  slice.
* Fixed: Text fields used a locale-aware validator while the underlying code did
  not.
* Fixed: Text fields will trigger a slice on text change, not only after focus
  change/enter press.
* Fixed: Rotate Tool snaps to incorrect value.
* Fixed: Object Collision would only moved objects to the right.
* Fixed: Object Collision would move the selected object when it should not.
* Fixed: Camera panning now works correctly instead of doing nothing.
* Fixed: Camera would flip around center point at maximum rotation.
* Fixed: Build platform grid blocked view from below objects.
* Fixed: Viewport on MacOSX with high-DPI screens was only taking 1/4th of the
window

Changes since 15.05.90
----------------------

* Fixed: Additional UI elements for tools and views not loading.
* Fixed: Double click needed to change setting dialog page.
* Fixed: Context menu entries (reload, center object, etc.) not working.
* Fixed: "Open With" or passing files from command line not working.
* Fixed: "Reload All" would not reload files.

In addition, a lot of work has gone into getting a usable Mac OSX version.

New Features
------------

* Plugin based system
  The Uranium framework provides us with a plugin-based system
  that provides additional flexibility when extending Cura. Think
  of new views, tools, file formats, etc. This is probably the
  biggest new feature.
* Improved UI
  The UI has received a complete overhaul.
* Time-Quality Slider
  The 4 static quick print profiles have been replaced with
  a slider that should make it easier to find the right spot
  between print time and print quality.
* More Settings
  The Advanced mode is now configurable and can show many
  additional settings that were previously not available, while at
  the same time not overwhelming new users with too many settings.
  Custom set of visible settings can be created by the user.
* Support for high-DPI screens
  The refreshed UI has been designed with high-DPI screens in
  mind which should improve the experience of Cura on such
  devices.
* Improved language support
  (Not yet available for the Beta release.)
* Improved support structure generation
  The new version of the CuraEngine now features improved
  support generation algorithms and additional options for support
  structure generation.
* Experimental Feature: Wire Printing
  Wire Printing has been added as an experimental new feature. It
  will print objects as a structure of lines. It can be enabled by
  from Advanced Mode -> Fixes -> Wire Printing.
* Undo/Redo
  It is now possible to undo and redo most scene operations, like
  moving or rotating objects.

Features from earlier versions not (yet) in this release
--------------------------------------------------------

* The All-at-once/One-at-a-time toggle is not available.
  We are working on an improved implementation of this mechanism
  but it will not be available for this release.
* No dual extrusion features are available yet.
  We are working on a completely new workflow for this but this
  needs additional time.

Cura/cura.py
#!/usr/bin/python
"""
This page is in the table of contents.
==Overview==
===Introduction===
Cura is a AGPL tool chain to generate a GCode path for 3D printing. Older versions of Cura where based on Skeinforge.
Versions up from 13.05 are based on a C++ engine called CuraEngine.
"""
__copyright__ = "Copyright (C) 2013 David Braam - Released under terms of the AGPLv3 License"

from optparse import OptionParser

from Cura.util import profile

def main():
	"""
	Main Cura entry point. Parses arguments, and starts GUI or slicing process depending on the arguments.
	"""
	parser = OptionParser(usage="usage: %prog [options] <filename>.stl")
	parser.add_option("-i", "--ini", action="store", type="string", dest="profileini",
		help="Load settings from a profile ini file")
	parser.add_option("-p", "--profile", action="store", type="string", dest="profile",
		help="Internal option, do not use!")
	parser.add_option("-s", "--slice", action="store_true", dest="slice",
		help="Slice the given files instead of opening them in Cura")
	parser.add_option("-o", "--output", action="store", type="string", dest="output",
		help="path to write sliced file to")
	parser.add_option("--serialCommunication", action="store", type="string", dest="serialCommunication",
		help="Start commandline serial monitor")

	(options, args) = parser.parse_args()

	if options.serialCommunication:
		from Cura import serialCommunication
		port, baud = options.serialCommunication.split(':')
		serialCommunication.startMonitor(port, baud)
		return

	print "load preferences from " + profile.getPreferencePath()
	profile.loadPreferences(profile.getPreferencePath())

	if options.profile is not None:
		profile.setProfileFromString(options.profile)
	elif options.profileini is not None:
		profile.loadProfile(options.profileini)
	else:
		profile.loadProfile(profile.getDefaultProfilePath(), True)

	if options.slice is not None:
		from Cura.util import sliceEngine
		from Cura.util import objectScene
		from Cura.util import meshLoader
		import shutil

		def commandlineProgressCallback(progress):
			if progress >= 0:
				#print 'Preparing: %d%%' % (progress * 100)
				pass
		scene = objectScene.Scene()
		scene.updateMachineDimensions()
		engine = sliceEngine.Engine(commandlineProgressCallback)
		for m in meshLoader.loadMeshes(args[0]):
			scene.add(m)
		engine.runEngine(scene)
		engine.wait()

		if not options.output:
			options.output = args[0] + profile.getGCodeExtension()
		with open(options.output, "wb") as f:
			gcode = engine.getResult().getGCode()
			while True:
				data = gcode.read()
				if len(data) == 0:
					break
				f.write(data)
		print 'GCode file saved : %s' % options.output

		engine.cleanup()
	else:
		from Cura.gui import app
		app.CuraApp(args).MainLoop()

if __name__ == '__main__':
	import os
	import sys

	# On Windows, the PATH variable can cause the search path for dlls
	# to give priority to dlls from other applications and it will cause
	# the bundled python dlls not to be loaded.
	# More specifically, anyone with Haskell Platform installed will not
	# be able to launch Cura because glut32.dll from Haskell is incompatible
	# with the bundled py-opengl and will cause Cura to crash
	if sys.platform.startswith('win'):
		os.environ['PATH'] = ''
	main()
